standardization/src/main/scala/za/co/absa/enceladus/standardization/interpreter/stages/TypeParser.scala

```diff
@@ -211,10 +223,60 @@ object TypeParser {
     override def assemblePrimitiveCastLogic: Column = column.cast(field.dataType)
   }
 
-  private final case class NumericParser(field: StructField,
-                                         path: String,
-                                         origSchema: StructType,
-                                         parent: Option[Parent]) extends ScalarParser
+  private trait NumericParser extends ScalarParser
+
+  private final case class IntegralParser(field: StructField,
+                                          path: String,
+                                          origSchema: StructType,
+                                          parent: Option[Parent],
+                                          overflowableTypes: Set[DataType ],
+                                          minValue: Long,
+                                          maxValue: Long
+                                         ) extends NumericParser {
+    override protected def assemblePrimitiveCastErrorLogic(castedCol: Column): Column = {
+      val basicLogic: Column = super.assemblePrimitiveCastErrorLogic(castedCol)
+
+      origType match {
+        case  dt: DecimalType =>
+          // decimal can be too big, to catch overflow or imprecision  issues compare to original
+          basicLogic or (column =!= castedCol.cast(dt))
+        case DoubleType | FloatType =>
+          // same as Decimal but directly comparing fractional values is not reliable,
+          // best check for whole number is considered modulo 1.0
+          basicLogic or (column % 1.0 =!= 0.0) or column > maxValue or column < minValue
+        case ot if overflowableTypes.contains(ot) =>
+          // from these types there is the possibility of under-/overflow, extra check is needed
+          basicLogic or (castedCol =!= column.cast(LongType))
+        case StringType => basicLogic or column.contains(".")// string of decimals are not allowed
+        case _ => basicLogic
```

Comment:
I would stick to keeping every implementation on a new line.

Reply:
I find it a prolongation of the code that does not bring extra clarity. 🤔

Reply:
Keeping one style is my point so that the reader doesn't have to switch